components/arm/fake/fake.go

```diff
@@ -113,7 +122,7 @@ func (a *Arm) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 
 	dof := len(model.DoF())
 	if dof == 0 {
-		a.logger.Info("fake arm built with zero degrees-of-freedom, nothing will show up on the Control tab " +
+		return errors.New("fake arm built with zero degrees-of-freedom, nothing will show up on the Control tab " +
```

Comment:
let's also expand the fake arm tests to check this reconfigured behavior and test one last time with the RC card.

Reply:
in trying to add a test I'm realizing this code is never reached. I think for there to be 0 dof, the model must be invalid so reconfigure exits with an error at an earlier point

Reply:
wait no, that's only true for `modelFromName`

Reply:
if you add a kinematics file with a zero dof arm, you can reach this code,

you can add an empty kinematics json to test this.